include_core/omrportsock.h

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(OMRPORTSOCK_H_)
+#define OMRPORTSOCK_H_
+
+#if defined(OMR_OS_WINDOWS) && !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif /* defined(OMR_OS_WINDOWS) && !defined(WIN32_LEAN_AND_MEAN) */
+
+#if defined(OMR_OS_WINDOWS) && defined(_WINSOCKAPI_)
+#undef _WINSOCKAPI_
+#endif /* defined(OMR_OS_WINDOWS) && defined(_WINSOCKAPI_) */
+
+#if defined(OMR_OS_WINDOWS)
+/* windows.h defined UDATA. Ignore its definition to avoid redefinition errors */
+#define UDATA UDATA_win32_
+#include <windows.h>
+#undef UDATA
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#pragma comment(lib, "Ws2_32.lib")
+#pragma comment (lib, "Mswsock.lib") 
+#pragma comment (lib, "AdvApi32.lib") 
+#else /* defined(OMR_OS_WINDOWS) */
+#include <sys/socket.h>
+#include <netinet/in.h>
+#endif /* defined(OMR_OS_WINDOWS) */
+
+#include<stdlib.h>
+#include<stdio.h>
+
+/* Common data types required for using the omr socket API */
+
+#if defined(OMR_OS_WINDOWS)
+typedef SOCKET OMRSocket;
+typedef SOCKADDR OMRSockAddr; /* for IPv4 */
+typedef SOCKADDR_IN OMRSockAddrIn;	/* for IPv4 addresses*/
+typedef SOCKADDR_IN6 OMRSockAddrIn6;  /* for IPv6 addresses*/
+typedef struct sockaddr_storage OMRSockAddrStorage; /* For IPv4 or IPv6 addresses */
+typedef struct addrinfoW OMRAddrInfo;  /* addrinfo structure – Unicode, for IPv4 or IPv6 */
+#else /* defined(OMR_OS_WINDOWS) */
+typedef int OMRSocket;
+typedef struct sockaddr OMRSockAddr;
+typedef struct sockaddr_in OMRSockAddrIn; /* for IPv4 addresses*/
+typedef struct sockaddr_in6 OMRSockAddrIn6; /* for IPv6 addresses*/
+typedef struct sockaddr_storage OMRSockAddrStorage; /* For IPv4 or IPv6 addresses */
```

Comment:
`OMRSockAddr` doesn't have enough storage to store IPv6 address though, where as `sockaddr_storage` does. `sockaddr_in` and `sockaddr_in6` are for creating a socket address structure that you can for instance pass into `bind()`, and other functions that needs a pointer to a socket address. Usually, you can cast `sockaddr_in` and `sockaddr_in6` pointers into and from `sockaddr` pointers interchangeably without a problem. However, the components  the structures are different. Sometimes, we don't use getaddrinfo() to get the address; we just create a `sockaddr_in` or a `sockaddr_in6` to directly fill in the socket address information ourselves. So I think we will need all 4 of these definitions because they are related but still different.

Reply:
sounds good.